### ISSUE TITLE
Extensions will be blocked if you do something that cost a lot of time in extension.

### ISSUE THREAD

**Issue by Lennon925** [microsoft/vscode#78416]
Currently, If you do something that cost a lot of time, info like 'cpu 100%' will occur and all extensions will be blocked. The feeling is so terrible.

How to solve this problem? Any suggestion for me?

**Comment by IllusionMH** [comment]
Do you know idea what extension causes this?
You can try steps from [Profile the running extensions](https://github.com/Microsoft/vscode/wiki/Performance-Issues#profile-the-running-extensions) or other tips from this page.

/needsMoreInfo

**Comment by alexandrudima** [comment]
Duplicate of #73041

**Comment by Lennon925** [comment]
> Duplicate of #73041

I don't think you have solved issue 73041. Why close this issue?

**Comment by alexandrudima** [comment]
@Lennon925 #73041 is closed with the resolution: "won't fix".

Our recommendation to extension authors is to not do expensive CPU operations in an extension because all extensions share a single extension host with a single event loop and rather spawn a separate process where they can do expensive stuff. You will find that the high quality extensions in our marketplace and that ship with VS Code respect this.

I respect your opinion, where you think the current architecture is stupid, and we will perhaps revisit our architecture in the future.

But to run extensions in different v8 isolates (in multiple threads) would mean to either multiply the memory used by text buffers by the number of extensions, or to implement a C++ immutable and thread-safe text buffer that could be exposed to multiple v8 isolates running in parallel. It would also mean we need to wait for workers to become stable in nodejs. It would expose issues with extensions that use native node modules which are not thread safe or with extensions that use each other's synchronous APIs, as those would still need to run on a single isolate.

In any case, we have arrived at the current architecture with a well understood and well considered trade-off between memory usage, implementation difficulty on our side, and pragmatism. 

You can create issues and comments, but at one point your comments on the same topic become noise that distract us from doing other important work instead of helping. Maybe you would be interested in contributing in another form, by tackling the problem, we do accept PRs as well. I would be very happy to have someone from the community contribute a text buffer implementation (for the extension host) that is written in C++ and that can be snapshotted and which can be accessed by multiple threads in parallel.
